### ISSUE TITLE
Indentation check gives false positive: 'array initialization rcurly' has incorrect indentation

### ISSUE THREAD

**Issue by rdiachenko** [checkstyle/checkstyle#8979]
Checkstyle 8.37, google-java-format 1.9
config (taken from google_checks.xml - https://github.com/checkstyle/checkstyle/blob/checkstyle-8.37/src/main/resources/google_checks.xml#L243):
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="Indentation">
      <property name="basicOffset" value="2"/>
      <property name="braceAdjustment" value="2"/>
      <property name="caseIndent" value="2"/>
      <property name="throwsIndent" value="4"/>
      <property name="lineWrappingIndentation" value="4"/>
      <property name="arrayInitIndent" value="2"/>
    </module>
  </module>
</module>
```
code:
```java
class Test {
  private static final String[] UNSECURED_PATHS = {
    "/health",
    "/version",
    "/metrics",
    "/docs",
    "/swagger*",
    "/webjars/springfox-swagger-ui/**",
    "/swagger-resources/**",
    "/openapi.json"
  };
}
```
Checkstyle gives indentation violation:
```bash
$ java -jar checkstyle-8.37-all.jar -c config.xml Test.java 
Starting audit...
[ERROR] /tmp/Test.java:11:3: 'array initialization rcurly' has incorrect indentation level 2, 
     expected level should be one of the following: 4, 8. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```
but there is an empty diff when applying google-java-format:
```bash
$ java -jar google-java-format-1.9-all-deps.jar Test.java > /tmp/FormattedTest.java \
   && diff -Naru Test.java /tmp/FormattedTest.java
$
```
**Note:** empty diff when applying google-java-format means it's not expected to have code formatted in a different way, however Indentation check gives false positive and requires the code to be formatted differently.

**Comment by toshetah** [comment]
@Abhishek-kumar09 , are you working on that? Or can I help with this one?

**Comment by Abhishek-kumar09** [comment]
@toshetah Thanks for your interest, The PR is under review process, I am actively looking for suggestions from mentors. I have done the required changes suggested by mentors. But they too need time to review it thoroughly so that nothing wrong gets deployed.

**Comment by rnveach** [comment]
Fixed was merged

**Comment by denis111** [comment]
I think it's not fixed or fixed incorrectly...

**Comment by Abhishek-kumar09** [comment]
@denis111 Could you provide a detailed explanation.
You can also create a new issue describing more about your problem https://github.com/checkstyle/checkstyle/issues/new?assignees=&labels=&template=bug_report.md&title=

**Comment by denis111** [comment]
@Abhishek-kumar09 
Well I'm not sure if it's a bug or a new behaviour. This code with Google style with `@ApiResponse` intentation 6 gives no warning with v8.36 but with v8.42 it wants indentation 8 or 10:
```java
  @Operation(summary = "Deletes a file from a transfer",
      description = "Deletes a file from a transfer", responses = {
      @ApiResponse(responseCode = OK_RESPONSE, description = OK_RESPONSE_DESCRIPTION),
```
Besides I have google style formatter plugin in my IDE and it formats it with intent of 6.

**Comment by Abhishek-kumar09** [comment]
To me what I found here that line 3 should follow lineWrapping indentation fundamentally since block is spanned across multiple lines.

Still if you want to have some different behaviour, please open a new issue and our team is happy to discuss the feasibility.

Will greatly appreciate.
